Blish HUD/GameServices/ArcDpsService.cs

```diff
@@ -1,96 +1,122 @@
 ﻿using System;
 using System.Diagnostics;
-using System.IO.Pipes;
+using System.Net;
 using System.Threading;
-using System.Threading.Tasks;
+using Blish_HUD.ArcDps;
+using Blish_HUD.ArcDps.Models;
 using Microsoft.Xna.Framework;
-using ProtoBuf;
 
 namespace Blish_HUD
 {
     public class ArcDpsService : GameService
     {
-        public bool ArcPresent;
-        private CancellationToken Cancel;
-        public bool HudIsActive;
-        private Mutex Mut;
-        private NamedPipeServerStream Server;
-        private Stopwatch Watch;
+        public delegate void RawCombatEvent(CombatEvent data);
+
+        private static readonly object WatchLock = new object();
+        private readonly TimeSpan _leeway = TimeSpan.FromMilliseconds(1000);
+        private bool _hudIsActive;
+#if DEBUG
+        public static long Counter = 0;
+#endif
+
+        private SocketListener _server;
+
+        private Stopwatch _stopwatch;
+
+        /// <summary>
+        ///     Indicates if arcdps updated <see cref="HudIsActive" /> in the last second (it should every in-game frame)
+        /// </summary>
+        public bool RenderPresent { get; private set; }
+
+        /// <summary>
+        ///     Indicates if arcdps currently draws its HUD (not in character select, cut scenes or loading screens)
+        /// </summary>
+        public bool HudIsActive
+        {
+            get
+            {
+                lock (WatchLock)
+                {
+                    return _hudIsActive;
+                }
+            }
+            private set
+            {
+                lock (WatchLock)
+                {
+                    _stopwatch.Restart();
+                    _hudIsActive = value;
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Holds unprocessed combat data
+        /// </summary>
+        public event RawCombatEvent OnRawCombatEvent;
```

Comment:
Can we have this be `EventHandler<RawCombatEventArgs>` where `RawCombatEventArgs` is a class that inherits from `EventArgs` with a property exposed for CombatEvent data?

Additionally (and I haven't been perfect at this - I've been fixing it when I see it), the event itself should not have "On" to prefix it.

Since this service won't be inherited and basically acts as a singleton, this is much less important, but normally, the method called to fire RawCombatEvent should be called `OnRawCombatEvent` and look like:

```
private void OnRawCombatEvent(RawCombatEventArgs e) {
RawCombatEvent?.Invoke(this, e);
}
```

Reply:
can you have a look now?

Reply:
Looks perfect!